### ISSUE TITLE
Handling of empty ByteBuffers in eta.runtime.io.MemoryManager

### ISSUE THREAD

**Issue by jneira** [typelead/eta#437]
While testing wai-servlet with the new version of eta (0.0.9) i got an error in the java helper code that converts `InputStream` to a `long` representing the address of a `ByteBuffer` in eta runtime memory manager (https://github.com/jneira/wai-servlet/blob/master/java/Utils.java#L24)
The error is a `NullPointerException` caused by `MemoryManager.getBoundedBuffer` returning null when you tries to get an empty ByteBuffer allocated previously with  `MemoryManager.allocateBuffer`

The simplest code to reproduce it:
```java
    public static void testEmptyBuffer () {
        System.out.println("Testing allocate an empty buffer with MemoryManager");
        byte[] bytes = new byte[] {};
        long address = MemoryManager.allocateBuffer(bytes.length,false);
        System.out.println("Address: "+address);
        ByteBuffer buf = MemoryManager.getBoundedBuffer(address);
        System.out.println("Buffer: "+buf);
    }
```
(executable example in https://github.com/jneira/eta-test/blob/master/java/Utils.java#L11)
The actual output is:
```
Testing allocate an empty buffer with MemoryManager
Address: 1
Buffer: null
```
i think the expected behaviour would be returning the empty buffer allocated at the supplied address.

**Comment by rahulmutt** [comment]
Thank you for the wonderfully detailed bug report :) It seems like the fix will involve statically allocating a zero-capacity ByteBuffer as shown here:

https://stackoverflow.com/questions/18223481/allocating-zero-capacity-bytebuffer

**Comment by jneira** [comment]
Maybe we can associate a static empty ByteBuffer with a fixed address to returning it directly?

**Comment by rahulmutt** [comment]
Yes, exactly what I was thinking. The encoding scheme with multi-size blocks in the new MemoryManager leaves several large gaps in the address space. I can select one of those gap addresses and we can make that the fixed value.

**Comment by jneira** [comment]
maybe it's is pointless but i thought in Integer.MAX_VALUE as its adress 
0 -> null, Integer.MAX_VALUE -> void buffer
Last possible address to represent an empty buffer has some logic for me.
do we have to take in account that to avoid take it by a ordinary buffer?

**Comment by jneira** [comment]
Anyway i am doing the changes using a constant for the address...

**Comment by rahulmutt** [comment]
Sounds good! Looking forward to the PR.

**Comment by rahulmutt** [comment]
This is fixed now with #438.

**Comment by jneira** [comment]
Yep, this one can be closed
